pex/util.py

```diff
@@ -198,21 +199,42 @@ def iter_pth_paths(filename):
     known_paths = set()
 
     with f:
-        for line in f:
+        for i, line in enumerate(f, start=1):
             line = line.rstrip()
             if not line or line.startswith("#"):
                 continue
             elif line.startswith(("import ", "import\t")):
+                # One important side effect of executing import lines can be alteration of the
+                # sys.path directly or indirectly as a programmatic way to add sys.path entries
+                # in contrast to the standard .pth mechanism of including fixed paths as
+                # individual lines in the file. Here we capture all such programmatic attempts
+                # to expand the sys.path and report the additions.
+                original_sys_path = sys.path[:]
                 try:
+                    # N.B.: Setting sys.path to empty is ok since all the .pth files we find and
+                    # execute have already been found and executed by our ambient sys.executable
+                    # when it started up before running this PEX file. As such, all symbols imported
+                    # by the .pth files then will still be available now as cached in sys.modules.
```

Comment:
Technically I think the .pth file could import some file that itself has a conditional import in code? Probably too esoteric to worry about, but might be worth a caveat in the comment if true.

Reply:
I've seen lots of conditional imports in .pth files, but all based on the version of the interpreter and that won't change here. In fact its hard to see what at all could be detected as different by the .pth between its execution 10ms ago duing sys.executable startup and here inside the same sys.executable. I think the only detectable difference is the fact we just blanked out sys.path.

Reply:
Similar - I think in the end unjustified - fears caused me to add the exception logging below though! Do you think that's enough?

Reply:
Yeah, I guess I was thinking of some weird corner case where we execute a conditional import in a function here, and it fails because we blanked out sys.path, but as you say, that would already have been executed during startup. Every line here has already been executed and its imports cached. So I think the logging is more than sufficient.